File: src/main/java/TimerView.java
import javax.swing.*;
import java.awt.event.ActionEvent;
import java.awt.event.ActionListener;
import java.util.concurrent.TimeUnit;

/**
 * Timer Model
 */

public class TimerView implements TimerContract.ITimerView {
    TimerContract.ITimerModel timer;
    JLabel timeDisplay;
    TimerContract.ITimerPresenter presenter;

    public static final int INITIAL_MINTUTES = 1;

    public static void main(String[] args) {
        TimerView view = new TimerView();
        view.setupSwing();
        view.presenter = new TimerPresenter(INITIAL_MINTUTES, view);
        view.updateTime(TimerPresenter.formatTime(INITIAL_MINTUTES * 60));
    }

    public void updateTime(String time) {
        timeDisplay.setText(time);
    }

    private void setupSwing() {
        JFrame frame = new JFrame();
        JButton start = new JButton("Start Timer");
        start.setBounds(150, 100, 100, 40);
        start.addActionListener(new ActionListener() {
            public void actionPerformed(ActionEvent e) {
                presenter.startTimer();
            }
        });
        JButton stop = new JButton("Stop Timer");
        stop.addActionListener(new ActionListener() {
            public void actionPerformed(ActionEvent e) {
                presenter.pauseTimer();
            }
        });
        stop.setBounds(150, 150, 100, 40);

        timeDisplay = new JLabel("00:00", SwingConstants.CENTER);
        timeDisplay.setBounds(150, 200, 100, 40);
        frame.add(timeDisplay);
        frame.add(start);
        frame.add(stop);
        frame.setSize(400, 500);
        frame.setLayout(null);
        frame.setVisible(true);
        frame.setDefaultCloseOperation(WindowConstants.EXIT_ON_CLOSE);
    }
}

File: src/main/java/TimerView.kt
import javax.swing.*
import javax.swing.text.SimpleAttributeSet
import javax.swing.text.StyleConstants
import javax.swing.text.StyledDocument
import java.awt.*
import java.awt.event.ActionEvent
import java.awt.event.ActionListener

/**
 * Timer Model
 */

class TimerView : TimerContract.ITimerView {
    internal var timeDisplay: JLabel = JLabel("00:00", SwingConstants.CENTER)
    internal val presenter: TimerContract.ITimerPresenter

    init {
        presenter = TimerPresenter(DEFAULT_MINUTES, this)
    }

    override fun updateTime(time: String) {
        timeDisplay.text = time
    }

    fun setupSwing() {
        val frame = JFrame()
        val start = JButton("Start Timer")
        start.setBounds(150, 100, 100, 40)
        start.addActionListener { presenter.startTimer() }
        val stop = JButton("Stop Timer")
        stop.addActionListener { presenter.pauseTimer() }
        stop.setBounds(150, 150, 100, 40)
        val reset = JButton("Reset Timer")
        val minuteBox = JTextPane()
        val doc = minuteBox.styledDocument
        val center = SimpleAttributeSet()
        StyleConstants.setAlignment(center, StyleConstants.ALIGN_CENTER)
        doc.setParagraphAttributes(0, doc.length, center, false)
        minuteBox.setBounds(150, 275, 100, 20)

        reset.setBounds(140, 200, 120, 40)
        reset.addActionListener {
            if (minuteBox.text == null) {
                presenter.resetTimer(DEFAULT_MINUTES)
            } else {
                presenter.resetTimer(Integer.valueOf(minuteBox.text)!!)
            }
        }

        timeDisplay = JLabel("00:00", SwingConstants.CENTER)
        timeDisplay.setBounds(150, 240, 100, 40)

        frame.add(timeDisplay)
        frame.add(start)
        frame.add(stop)
        frame.add(minuteBox)
        frame.add(reset)
        frame.setSize(400, 500)
        frame.layout = null
        frame.isVisible = true
        frame.defaultCloseOperation = WindowConstants.EXIT_ON_CLOSE
    }
}

val DEFAULT_MINUTES = 3